### PR TITLE
Issue 7, time synchronization:  add -k,--clockid options

### PR DESCRIPTION
**Pull request by phijor**

### CODE REVIEW

File: src/config.cpp [hunk 1]
```diff
@@ -190,6 +193,50 @@ void parse_program_options(int argc, const char** argv)
...
+            },
+            {
+                "boottime", CLOCK_BOOTTIME,
+            },
+        };
+
+        const char* clock_name_ptr = clock_name.c_str();
+        for (auto clock : clocks)
+        {
+            if (std::strcmp(clock_name_ptr, clock.name) == 0)
```
Comment (bmario) [inline src/config.cpp]:
```
if (clock_name == clock.name)
```

File: src/config.cpp [hunk 2]
```diff
@@ -190,6 +193,50 @@ void parse_program_options(int argc, const char** argv)
...
     }
 
+    // time synchronization
+    config.use_clockid = false;
+    if (!clock_name.empty())
+    {
+#if defined(USE_PERF_CLOCKID) && !defined(HW_BREAKPOINT_COMPAT)
+        struct clock_descripton
+        {
+            const char name[16];
```
Comment (bmario) [inline src/config.cpp]: Is there a reason, why this isn't a std::string?

File: src/config.cpp [hunk 3]
```diff
@@ -190,6 +193,50 @@ void parse_program_options(int argc, const char** argv)
...
+            },
+            {
+                "realtime", CLOCK_REALTIME,
+            },
+            {
+                "boottime", CLOCK_BOOTTIME,
+            },
+        };
+
+        const char* clock_name_ptr = clock_name.c_str();
```
Comment (bmario) [inline src/config.cpp]: Not needed anymore after the next suggestion.

File: src/config.cpp [hunk 4]
```diff
@@ -79,6 +79,8 @@ void parse_program_options(int argc, const char** argv)
     bool kernel, no_kernel;
     std::uint64_t read_interval_ms;
 
+    std::string clock_name;
```
Comment (bmario) [inline src/config.cpp]: Maybe rename to `requested_clock` or `requested_clock_name`

File: src/config.cpp [hunk 5]
```diff
@@ -110,6 +112,7 @@ void parse_program_options(int argc, const char** argv)
              "enable global recording of a raw tracepoint event (usually requires root)")
         ("metric-event,E", po::value(&config.perf_events),
              "the name of a perf event to measure") // TODO: optionally list available events
+        ("clockid,k", po::value(&clock_name), "select timer to use for timestamps")
```
Comment (bmario) [inline src/config.cpp]: I'd change the description to something like:

"clock used for perf timestamps"

File: src/config.cpp [hunk 6]
```diff
@@ -190,6 +193,50 @@ void parse_program_options(int argc, const char** argv)
...
+            {
+                "realtime", CLOCK_REALTIME,
+            },
+            {
+                "boottime", CLOCK_BOOTTIME,
+            },
+        };
+
+        const char* clock_name_ptr = clock_name.c_str();
+        for (auto clock : clocks)
```
Comment (bmario) [inline src/config.cpp]:
```
for (const auto& clock : clocks)
```

File: src/config.cpp (hunk 2 above)
Comment (phijor) [reply src/config.cpp]: I wanted the struct to be usable as `constexpr`, that won't work with `std::string`s. But even then I'd think `const char*` might be better than `const char[16]` :thinking:

But if you think using `std::string` is more idiomatic I'll change it.

Comment (bmario) [reply src/config.cpp]: While `constexpr` certainly sounds like a valid point, it smells like premature optimization for me. 
`clock_description` is static anyways, so I don't see the benefit for `constexpr` here.
Using std::string is just less demanding in terms of brain computing capacity.

**Pull request merged**